scripts/onsite/onsite_create_calibration_file.py

```diff
@@ -0,0 +1,120 @@
+#!/usr//bin/env python
+
+"""
+ F. Cassol, 12/11/2019
+
+ Onsite script for creating a flat-field calibration file file to be run as a command line:
+
+ --> onsite_create_calibration_file
+
+"""
+
+import argparse
+from pathlib import Path
+from lstchain.io.data_management import *
+import lstchain.visualization.plot_calib as calib
+
+# to be changed with the right one : /fefs/aswg/data/real
+base_dir = '/ctadata/franca/fefs/aswg/real'
+#base_dir = '/fefs/aswg/data/real'
+
+# parse arguments
+parser = argparse.ArgumentParser(description='Create flat-field calibration files',
+                                 formatter_class = argparse.ArgumentDefaultsHelpFormatter)
+required = parser.add_argument_group('required arguments')
+optional = parser.add_argument_group('optional arguments')
+
+required.add_argument('-r', '--run_number', help="Run number if the flat-field data",
+                      type=int, required=True)
+required.add_argument('-p', '--pedestal_run', help="Run number of the drs4 pedestal run",
+                      type=int, required=True)
+
+optional.add_argument('-v', '--version', help="Version of the production",
+                      type=int, default=0)
+optional.add_argument('-s', '--statistics', help="Number of events for the flat-field and pedestal statistics",
+                      type=int, default=10000)
+
+args = parser.parse_args()
+run = args.run_number
+ped_run = args.pedestal_run
+prod_id = 'v%02d'%args.version
+stat_events = args.statistics
+max_events = 1000000
+
+
+def main():
+
+    print(f"\n--> Start calculating calibration from run {run}")
+
+    try:
+        # verify input file
+        file_list=sorted(Path(f"{base_dir}/R0").rglob(f'*{run}.0000*'))
```

Comment:
What about run `LST-1.2.Run01541.0040.fits.fz` for example ?

Reply:
The statistics of one subrun is  more than enough so we need to read only the 0000 run

Reply:
Yes but then the script does not do the intended since user pass `--statistics` thinking it will use this value.
Could you add a warning if `max_events<stat_events` and if, at the end of the process, a number of events lower that `stat_events` have been used, please?